src/NorthwindWebsite/API/Controllers/CategoriesController.cs

```diff
@@ -23,4 +26,41 @@ public async Task<ActionResult<List<Category>>> GetCategories()
 
         return categories.ToList();
     }
+
+    [HttpGet("{categoryId}/image")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult> GetImage(int categoryId)
+    {
+        var picture = await _categoryService.GetImage(categoryId);
+
+        if (picture == null)
+        {
+            return NotFound();
+        }
+
+        return File(picture, FileFormat);
+    }
+
+
+    [HttpPut("{categoryId}/image")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult> UploadImage(
+        int categoryId,
+        [FromBody] ImageDto imageDto)
+    {
+        var category = await _categoryService.Get(categoryId);
+
+        if (category == null)
+        {
+            return BadRequest();
```

Comment:
I am not sure that sending `400` is good here: in this case you are just not able to find particular entity, not handle it. It's better to return `404` here and also add some descriptive message kinda `Cannot find category by provided id`